src/helpers/recompute_zones.ts

```diff
@@ -0,0 +1,398 @@
+ * value is a sorted number array representing a
+ * profile.
+ * 
+ * 
+ * See the following example:    here profileStartingPosition    
+ *                               corresponds to [A,C,E,G,K]             
+ *    A B C D E F G H I J K      so with number [0,2,4,6,10]              
+ *  1    '   '   '       '                                           
+ *  2    '   '   '_______'       here profile correspond            
+ *  3    '___'   |_______|       for A to []                         
+ *  4    |   |                   for C to [3, 5]                       
+ *  5    |___|                   for E to []                            
+ *  6                            for G to [2, 3]                        
+ *  7                            for K to []                               
+ *
+ * 
+ * Now we can easily find which zones are overlapping
+ * with a given zone. Suppose we want to add a new zone
+ * ["D5:H6"] to the ensemble:                                             
+ *                                                                    
+ *                              With a binary search of left and right                                                 
+ *    A B C D E F G H I J K     on profilesStartingPosition, we can                    
+ *  1    '   '   '       '      find the indexes of the profiles on which          
+ *  2    '   '   '_______'      apply a modification.                     
+ *  3    '___'   |_______|                                                                      
+ *  4    |  _|_______           Here we will:                                        
+ *  5    |_|_|       |          - add a new profile in D                          
+ *  6      |_________|          - modify the profile in E                                 
+ *  7                           - modify the profile in G                        
+ *                              - add a new profile in I                           
+ *                                         
+ *  See below the result:
+ *                              
+ *                              Note the particularity of the profile                                                                    
+ *    A B C D E F G H I J K     for G: it will correspond to [2, 3, 4, 6]                   
+ *  1    ' ' '   '   '   '                                                         
+ *  2    ' ' '   '___'___'      For know how to modify the profile (add a       
+ *  3    '_'_'   |___|___|      filled zone or remove it) we do a binary          
+ *  4    | | |___ ___           search of the top and bottom value on the        
+ *  5    |_| |   |   |          profile array. Function of the result index       
+ *  6      |_|___|___|          is odd or even, we know if we are in a           
+ *  7                           'filled' zone or not and how operate.                
+ *                                                            
+ * /
```

Comment:
```suggestion
 */
```
this will fix the issue Adrien mentions below ^^

Reply:
you are a good man <3